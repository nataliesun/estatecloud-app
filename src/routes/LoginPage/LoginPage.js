import React, { Component } from 'react';
import LoginForm from '../../components/LoginForm/LoginForm';

import './LoginPage.scss'

class LoginPage extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => { },
    },
  }

  handleLoginSuccess = () => {
    const { location, history } = this.props
    const destination = (location.state || {}).from || '/dashboard'
    history.push(destination)
  }

  render() {
    return (
      <div className="LoginPage">
        <h2>Login</h2>
        <LoginForm
          onLoginSucess={this.handleLoginSuccess}
          {...this.props}
        />
      </div>
    );
  }
}

export default LoginPage;