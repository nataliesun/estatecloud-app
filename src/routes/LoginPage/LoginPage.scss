@import '../../components/variables.scss';

.LoginPage {
  color: $grey-color;
   display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: $medium-space;
}