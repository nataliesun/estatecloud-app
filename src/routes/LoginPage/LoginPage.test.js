import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import LoginPage from './LoginPage';

describe('LoginPage route component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <LoginPage />
      , div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <LoginPage />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})