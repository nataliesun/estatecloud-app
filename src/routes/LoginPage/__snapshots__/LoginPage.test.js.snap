// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage route component renders the UI as expected 1`] = `
<div
  className="LoginPage"
>
  <h2>
    Login
  </h2>
  <form
    className="LoginForm"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div
      className="email"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        autoComplete="new-password"
        id="email"
        name="email"
        required={true}
        type="email"
      />
    </div>
    <div
      className="password"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        autoComplete="new-password"
        id="password"
        name="password"
        required={true}
        type="password"
      />
    </div>
    <button
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
